### PR TITLE
[IOPAE-312] `OnServiceLifecycleChange` entrypoint setup 

### PR DESCRIPTION
**Pull request by balanza**
#### List of Changes

A series of utilities to map domain processing to azure functions runtime.

#### Motivation and Context

We have to handle the reads on `ServiceLifecycle`'s change feed. Changes are red on batches. This PR proposes a series of utilities that maps a single document processor over a batch of documents.

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by rodoviti** [comment]
@balanza I didn't understand the `setBindings` function! 😵‍💫

**Comment by balanza** [comment]
> @balanza I didn't understand the `setBindings` function! 😵‍💫

It takes a function in the form
```ts
fn(value: SomeValue) => Record<BindingNames, unknown>
```
Where `SomeValue` is the result of previous operations and `BindingNames` is the set of keys that identifies output bindings to be set. In plain English, the function would be
```
Tell me what you're gonna assign to bindings, taking SomeValue as incoming data
```
The `setBindings` function then maps `fn` over a `ReaderTaskEither` (which is "a `TaskEither` that takes arguments") to enable composability with other parts of the procedure.


Using the use case of this PR as an example, we can tell `setBindings` as
```
We process the batch of documents and get either an error or a set of jobs to be spawned via queue messages.
Messages are sent by assigning an array of messages to the relative context.bindings attribute.
setBindings is in charge of such assignment if and only the procedure is not in error. It takes a functor to compose the value to be assigned to bindings, using the result of the procedure as input data.


```

----
(this is a draft, I'm gonna document it with comments and tests in the next commits)

File: apps/io-services-cms-webapp/src/lib/azure/__tests__/adapters.test.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  });
+
+  it("should fail on failing procedure", async () => {
+    const aFailingProcedure = vi.fn(() => TE.left("ko"));
+
+    const handler = toAzureFunctionHandler(aFailingProcedure);
+
+    try {
+      const result = await handler(mockContext);
+      shouldNotBeHere(result);
```
[the remark this answers is not in the document]
Comment (balanza) [reply apps/io-services-cms-webapp/src/lib/azure/__tests__/adapters.test.ts:56]: You are right, it's just me not knowing `vitest` good enough 😅 

The meaning is "the execution is not supposed to pass here, something is wrong in your test".

I'm gonna use `assert.fail` as you suggested, thanks

[the remark this answers is not in the document]
Comment (balanza) [reply apps/io-services-cms-webapp/src/lib/azure/__tests__/adapters.test.ts]: done in https://github.com/pagopa/io-services-cms/pull/51/commits/a57cd623d664e9097674fa1f1a2a9afa76017d45

**Review by rodoviti — APPROVED** [review]
